File: everything2-zipfile-processor/handler.rb
#!/usr/bin/env ruby
#frozen_string_literal: true

require 'aws-sdk-s3'
require 'archive-zip'

def http_response(code, message)
  {"statusCode": code, "headers": {"Content-Type": "application/json"}, "body": {"message": message}.to_json}
end

def lambda_handler(args)
  s3client = Aws::S3::Client.new(region: ENV['AWS_DEFAULT_REGION'])
  event = args[:event]

  pp event

  return http_response(200, "OK")
end

File: perl-layer-test/handler.pl
#!./bin/perl -w

use strict;

sub http_response
{
  my ($code, $message) = @_;
  return encode_json({"statusCode" => $code, "headers" => {"Content-Type" => "application/json"}, "body" => {"message" => $message}});
}

sub lambda_handler
{
  my ($event) = @_;
  http_response(200, "Hello World - Layer test!");
}

1;
